src/application_settings/type_notation_helper.py

```diff
@@ -1,17 +1,20 @@
-# pylint: disable=consider-alternative-union-syntax
+# pylint: disable=consider-alternative-union-syntax, useless-suppression
 """Defines type aliasses that handle notational differences between python versions."""
 import sys
 from pathlib import Path
+from typing import Any
 
 if sys.version_info >= (3, 10):
     from typing import TypeAlias
 
     PathOrStr: TypeAlias = Path | str
+    dictOrAny: TypeAlias = dict[str, Any] | Any  # pylint: disable=invalid-name
     PathOpt: TypeAlias = Path | None
 else:
-    from typing import Optional, Union
+    from typing import Union
 
     from typing_extensions import TypeAlias
 
     PathOrStr: TypeAlias = Union[Path, str]
-    PathOpt: TypeAlias = Optional[Path]
+    dictOrAny: TypeAlias = Union[dict[str, Any], Any]  # pylint: disable=invalid-name
```

Comment:
Do we not avoid the `invalid-name` error if we make this `DictOrAny`?

Reply:
We actually don't use it anymore, removed.